### DISCUSSION TITLE
Simplified uninstallation of various setup variants with automatic determination of the uninstaller and manufacturer and version checks

### DISCUSSION THREAD

**Discussion by StHoBT** [PSAppDeployToolkit/PSAppDeployToolkit#770] [Ideas]
I have developed an uninstall routine that determines the programs to be uninstalled and the corresponding uninstall call in the registry under 'Add- Remove Programs' (32 and 64Bit) based on the app name, if applicable, the manufacturer and the version. By passing the specific 'Silent' parameter, the program(s) will be silently uninstalled.
Detailed logs are also written.

I can provide the complete function here.

Here is the function description:

Function Remove-Application**s**{
<#
.SYNOPSIS
	Uninstall Applications listed in Add-Remove-Programs
.DESCRIPTION
	with this function is it possible to find and uninstall a Application with the original uninstall-parameters and in Silent-Mode.
.SYNTAX
    Remove-Applications [-appName] <String> [[-appVendor] <String>] [[-appVersion] <String>] [[-SilentParameter] <String>] [[-RemoveOriginalUninstallParameter]<switch>] [[-IgnoreExitCodes] <String>] [[-ContinueOnError] <boolean>] [[-PassThru] <boolean>] [[-Debuglevel] <Int>] 
.PARAMETER appName
	the Characters the Application-Name starts with. The Parameter is Mandatory. appName="Android"
.PARAMETER appVendor
	the Characters the Vendor of the Application-Name starts with. The Parameter is optional. appVendor="Googl"
.PARAMETER appVersion
	the Characters the DisplayVersion of the Application-Name starts with. The Parameter is optional. appVersion="12.07"
.PARAMETER SilentParameter
	the Parameter to uninstall the Software in SilendMode. The Parameter is optional. If not set the uninstallation will not be silent. SilentParameter="/Silent"
.PARAMETER RemoveOriginalUninstallParameter
	Removes all existing UninstallParameters "Uninstall.exe" /U /S /X -> "Uninstall.exe"
.PARAMETER ReplaceSomeOriginalUninstallParameter
	Replace some Uninstall Parameters  "$find>$Repalce","$find>$Repalce"  The '>' is the seperator between find and replace string
	"/U>/S","C:>D:"
.PARAMETER Debuglevel
	if the debuglevel is higher then 0 debuginformations will be written in the 'Bertrandt Custom Log'. The Parameter is optional. default=0
	0=no debug , 1=only ReturnValue , 2=full debug with most values
.PARAMETER IgnoreExitCodes
	List the exit codes to ignore.
.PARAMETER ContinueOnError
	Continue if an exit code is returned by the process that is not recognized by the App Deploy Toolkit. Default: $false.
.PARAMETER PassThru
	Returns ExitCode, STDOut, and STDErr output from the process.
.PARAMETER NoWait
	Immediately continue after executing the process.
.EXAMPLE 
	Remove-Applications "Android" "Googl" "12." "/Silent"
.EXAMPLE
	Remove-Applications "Android" "Googl" 2  -> no silent uninstall
.EXAMPLE
	Remove-Applications -appName "Android" -appVendor "Googl" -appVersion "12.07" -SilentParameter "/S /Z" -Debuglevel 2
.EXAMPLE
	Remove-Applications -appName "Android" -appVendor "Googl" -appVersion "12.07" -SilentParameter "/S /Z" -Debuglevel 2 -IgnoreExitCodes '1,2'
.NOTES
	Attention: You must find out the Silent-Parameter by yourself and put it in the SilentParameter.
	Attention: If you chose -appName "Microsoft" -appVendor "Microsoft" as Parameter the most Microsoft-Software will be uninstalled
	Author of this feature: Steffen Hoffmann
#>

**Comment by cyborgrd** [comment]
Please use "Add Code" after copy /paste and marking, otherwise your scripts are broken.

**Comment by StHoBT** [comment]
I’m sorry.

I’m new at the forum and I never done something with Git.

I will remember next Time.


Thx


Von: cyborgrd ***@***.***>
Gesendet: Dienstag, 21. Februar 2023 15:36
An: PSAppDeployToolkit/PSAppDeployToolkit ***@***.***>
Cc: Hoffmann, Steffen ***@***.***>; Author ***@***.***>
Betreff: [EXTERNAL] Re: [PSAppDeployToolkit/PSAppDeployToolkit] Simplified uninstallation of various setup variants with automatic determination of the uninstaller and manufacturer and version checks (Discussion #770)


Please use "Add Code" after copy /paste and marking, otherwise your scripts are broken.

**Reply by andrewhiz** [reply]
@StHoBT  - Would love to see your code. Do you mind sharing it as an extension so the community can reffer to it?
You can add here - https://discourse.psappdeploytoolkit.com/c/thetoolkit/extensions/13

Or you can add it as a comment here, make sure to use the code snippet to past your code. Simply type "/" and choose code block.

**Comment by That-Annoying-Guy** [comment]
FYI: You can also post custom extensions here: https://discourse.psappdeploytoolkit.com/c/thetoolkit/extensions/13

Just remember to use the <\> tool for code.

**Comment by mjr4077au** [comment]
Similar functionality has been incorporated into PSADT v4.
